src/java/org/apache/cassandra/index/sasi/conf/view/RangeTermTree.java

```diff
@@ -28,77 +28,65 @@
 import org.apache.cassandra.index.sasi.plan.Expression;
 import org.apache.cassandra.db.marshal.AbstractType;
 import org.apache.cassandra.utils.Interval;
-import org.apache.cassandra.utils.IntervalTree;
+import org.apache.cassandra.utils.IntervalList;
 
-public class RangeTermTree implements TermTree
-{
+public class RangeTermTree implements TermTree {
     protected final ByteBuffer min, max;
-    protected final IntervalTree<Term, SSTableIndex, Interval<Term, SSTableIndex>> rangeTree;
+    protected final IntervalList<Term, SSTableIndex, Interval<Term, SSTableIndex>> rangeTree;
     protected final AbstractType<?> comparator;
 
-    public RangeTermTree(ByteBuffer min, ByteBuffer max, IntervalTree<Term, SSTableIndex, Interval<Term, SSTableIndex>> rangeTree, AbstractType<?> comparator)
-    {
+    public RangeTermTree(ByteBuffer min, ByteBuffer max,
+            IntervalList<Term, SSTableIndex, Interval<Term, SSTableIndex>> rangeTree, AbstractType<?> comparator) {
         this.min = min;
         this.max = max;
         this.rangeTree = rangeTree;
         this.comparator = comparator;
     }
 
-    public Set<SSTableIndex> search(Expression e)
-    {
+    public Set<SSTableIndex> search(Expression e) {
         ByteBuffer minTerm = e.lower == null ? min : e.lower.value;
         ByteBuffer maxTerm = e.upper == null ? max : e.upper.value;
 
-        return new HashSet<>(rangeTree.search(Interval.create(new Term(minTerm, comparator),
-                                                              new Term(maxTerm, comparator),
-                                                              (SSTableIndex) null)));
+        return new HashSet<>(rangeTree.search(
+                Interval.create(new Term(minTerm, comparator), new Term(maxTerm, comparator), (SSTableIndex) null)));
     }
 
-    public int intervalCount()
-    {
+    public int intervalCount() {
         return rangeTree.intervalCount();
     }
 
-    static class Builder extends TermTree.Builder
-    {
+    static class Builder extends TermTree.Builder {
```

Comment:
Our static analysis engine detects that your pull request includes a change that **class name shadows simple name of superclass**. According to [the best practice](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-class-names-shouldn-t-shadow-simple-name-of-superclass-nm-same-simple-name-as-superclass), it can be exceptionally confusing, creating many situations where you have to look at import statements to resolve references, and creating many opportunities to accidentally define methods that don't override the methods in their superclasses.